app/assets/javascripts/discourse/app/components/ace-editor.js

```diff
@@ -123,8 +123,12 @@ export default class AceEditor extends Component {
             });
             editor.getSession().setMode("ace/mode/" + this.mode);
             editor.on("change", () => {
-              this._skipContentChangeEvent = true;
-              this.set("content", editor.getSession().getValue());
+              if (this.onChange) {
+                this.onChange(editor.getSession().getValue());
+              } else {
+                this._skipContentChangeEvent = true;
+                this.set("content", editor.getSession().getValue());
+              }
```

Comment:
I needed to have an onChange event for the code component, everything in FormKit is following DDAU, there's no implicit mutation